CONTRIBUTING.md

```diff
@@ -0,0 +1,334 @@
+this means the lines that you changed were also changed after you created your branch and in this
+case git does not know what the right change is. You will need to manually resolve it, check
+[here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line)
+for more information on how to do this.
+
+It is recommended to always rebase on a new push to ensure it is testing against the latest code.
+
+## Rust Dependency updates
+
+To automatically keep dependencies up to date we use the [renovate](https://github.com/renovatebot/renovate) bot.
+The bot automatically opens new PRs with updates that should be merged by maintainers.
+
+However sometimes, especially during development, it can be the case that you like to update a dependency.
+
+To do this, you can either run `cargo update` (to update all dependencies) or change the version of the specific dependency you want to update in `Cargo.toml`.
+
+Please run `make` after this to ensure the project still builds after your dependency updates.
+It may be necessary to make code changes to address the updates dependencies.
+
+Then commit the changes and open a PR. If you want to add other changes it is recommended to keep the
+dependency updates in their own commit as this makes reviewing them much easier.
+
+## Find bad changes with git bisect
+
+git bisect is very powerful command in order to quickly find commits that caused a regression.
+
+For example assume you did a Podman update and now something that used to work fine is no longer working,
+this is called a regression. If the change was not intentional it may be hard to find out what caused it.
+git bisect can help with that.
+
+First you need to know the last working version and the the new version were is stopped working and you
+should have a simple test for you behavior. Then run
+```
+$ git bisect start <bad version> <good version>
+```
+
+Now git will go through the commits between them via binary search to find the first bad commit.
+You need to compile the binary, then do your test and see if this works or not then use
+```
+$ git bisect good
+```
+if is is working or if it is not working:
+
+```
+$ git bisect bad
+```
+
+Then again compile and test and repeat the steps until git found only one commit left, this
+should be the first bad commit. If you file an issue this information is very useful to us
+developers to quickly see the root cause.
+
+Given this can be a long manual process you can automate the bisect run if you have a good reproducer.
+For example lets assume there is regression with `podman run $IMAGE someCommand` where it fails to
+run and throws and error.
+You can automate this after the bisect start command to give the good and bad version by using
+```
+$ git bisect run sh -c "make podman && bin/podman run $IMAGE someCommand || exit 1"
```

Comment:
It would be good to have a chapter on how I import my changes into the podman. I would expect a similar chapter like this  https://github.com/containers/common/blob/main/CONTRIBUTING.md#test-changes-in-a-dependent-repository